When overridding logging handlers for a specific category we get NullPointerException in native mode only

**Describe the bug**
Whenever we configure specific handlers that we link to a category we get a NullPointerException.
```
2020-01-30 12:10:53 Mounting /private/var/folders/xk/xtn2b68s2c33qryl26pvzfv40000gn/T/tmpt7c35ugv as /var/task:ro,delegated inside runtime container
Exception in thread "main" java.lang.RuntimeException: Failed to start quarkus
at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:138)
at io.quarkus.runtime.Application.start(Application.java:87)
at io.quarkus.runtime.Application.run(Application.java:210)
at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:41)
Caused by: java.lang.NullPointerException
at org.jboss.logmanager.AtomicArray.add(AtomicArray.java:114)
at org.jboss.logmanager.LoggerNode.addHandler(LoggerNode.java:295)
at org.jboss.logmanager.Logger.addHandler(Logger.java:217)
at io.quarkus.runtime.logging.LoggingSetupRecorder.addNamedHandlersToCategory(LoggingSetupRecorder.java:169)
at io.quarkus.runtime.logging.LoggingSetupRecorder.initializeLogging(LoggingSetupRecorder.java:116)
at io.quarkus.deployment.steps.LoggingResourceProcessor$setupLoggingRuntimeInit180.deploy_0(LoggingResourceProcessor$setupLoggingRuntimeInit180.zig:103)
at io.quarkus.deployment.steps.LoggingResourceProcessor$setupLoggingRuntimeInit180.deploy(LoggingResourceProcessor$setupLoggingRuntimeInit180.zig:122)
at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:60)
... 3 more
```

**Expected behavior**
The behaviour should be the same as when running in JVM and attach the extra handlers to the Logger that belongs to a category.

**To Reproduce**
Steps to reproduce the behavior:
1. Configure named handlers and link to category
2. Build native image
3. During initialisation the exception occurs

**Configuration**
```properties
quarkus.log.handler.console."STRUCTURED_LOGGING".format=%m%n

quarkus.log.category."com.xyz.DocumentEventLogger".level=DEBUG
quarkus.log.category."com.xyz.DocumentEventLogger".handlers=STRUCTURED_LOGGING
```

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Darwin APRXJGH6D5D5F9 18.7.0 Darwin Kernel Version 18.7.0: Thu Jan 23 06:52:12 PST 2020; root:xnu-4903.278.25~1/RELEASE_X86_64 x86_64
 - Output of `java -version`: 1.8.0_211
 - GraalVM version (if different from Java): quay.io/quarkus/ubi-quarkus-native-image:19.2.1
 - Quarkus version or git rev: 1.2.0.Final

Getting a similar (almost identical) dump on startup for native-image running `1.2.0-Final`

```
target/lunar-lander-1.0.0-SNAPSHOT-runner 
Exception in thread "main" java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:194)
	at io.quarkus.runtime.Application.start(Application.java:87)
	at io.quarkus.runtime.Application.run(Application.java:210)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:41)
Caused by: java.lang.NullPointerException
	at io.quarkus.runtime.logging.LoggingSetupRecorder.createNamedHandlers(LoggingSetupRecorder.java:138)
	at io.quarkus.runtime.logging.LoggingSetupRecorder.initializeLogging(LoggingSetupRecorder.java:105)
	at io.quarkus.deployment.steps.LoggingResourceProcessor$setupLoggingRuntimeInit10.deploy_0(LoggingResourceProcessor$setupLoggingRuntimeInit10.zig:103)
	at io.quarkus.deployment.steps.LoggingResourceProcessor$setupLoggingRuntimeInit10.deploy(LoggingResourceProcessor$setupLoggingRuntimeInit10.zig:122)
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:84)
```

@dmlloyd, sure, my [minimal working example](https://github.com/schulzp/quarkus-logging-npe) uses quarkus 1.2.0 and right after running the -runner.jar [inside a container] with [Open]JDK 11 the error occurs:

```text
Exception in thread "main" java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:210)
	at io.quarkus.runtime.Application.start(Application.java:87)
	at io.quarkus.runtime.Application.run(Application.java:210)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:41)
Caused by: java.lang.NullPointerException
	at io.quarkus.runtime.logging.LoggingSetupRecorder.createNamedHandlers(LoggingSetupRecorder.java:138)
	at io.quarkus.runtime.logging.LoggingSetupRecorder.initializeLogging(LoggingSetupRecorder.java:105)
	at io.quarkus.deployment.steps.LoggingResourceProcessor$setupLoggingRuntimeInit2.deploy_0(LoggingResourceProcessor$setupLoggingRuntimeInit2.zig:103)
	at io.quarkus.deployment.steps.LoggingResourceProcessor$setupLoggingRuntimeInit2.deploy(LoggingResourceProcessor$setupLoggingRuntimeInit2.zig:122)
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:68)
	... 3 more
```

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -157,7 +157,7 @@
         <maven-invoker.version>3.0.1</maven-invoker.version>
         <awaitility.version>4.0.3</awaitility.version>
         <jprocesses.version>1.6.5</jprocesses.version>
-        <jboss-logmanager.version>1.0.6</jboss-logmanager.version>
+        <jboss-logmanager.version>1.0.9</jboss-logmanager.version>
         <jgit.version>5.11.0.202103091610-r</jgit.version>
         <flyway.version>7.7.0</flyway.version>
         <yasson.version>1.0.8</yasson.version>

--- a/core/runtime/src/main/java/io/quarkus/runtime/graal/LoggingSubstitutions.java
+++ b/core/runtime/src/main/java/io/quarkus/runtime/graal/LoggingSubstitutions.java
@@ -1,5 +1,6 @@
 package io.quarkus.runtime.graal;
 
+import java.util.concurrent.atomic.AtomicReference;
 import java.util.logging.Handler;
 
 import org.jboss.logmanager.LogContext;
@@ -20,8 +21,8 @@ import io.quarkus.bootstrap.logging.QuarkusDelayedHandler;
 final class Target_org_jboss_logmanager_LoggerNode {
 
     @Alias
-    @RecomputeFieldValue(kind = RecomputeFieldValue.Kind.Reset)
-    volatile Handler[] handlers;
+    @RecomputeFieldValue(declClass = AtomicReference.class, kind = RecomputeFieldValue.Kind.NewInstance)
+    AtomicReference<Handler[]> handlersRef;
 }
 
 @TargetClass(className = "org.slf4j.LoggerFactory")

--- a/independent-projects/bootstrap/pom.xml
+++ b/independent-projects/bootstrap/pom.xml
@@ -42,7 +42,7 @@
         <commons-lang.version>3.12.0</commons-lang.version>
         <guava.version>30.1-jre</guava.version>
         <shrinkwrap-depchain.version>1.2.6</shrinkwrap-depchain.version>
-        <jboss-logmanager-embedded.version>1.0.5</jboss-logmanager-embedded.version>
+        <jboss-logmanager-embedded.version>1.0.9</jboss-logmanager-embedded.version>
         <slf4j-jboss-logmanager.version>1.1.0.Final</slf4j-jboss-logmanager.version>
         <slf4j-api.version>1.7.30</slf4j-api.version>
         <graal-sdk.version>21.0.0</graal-sdk.version>

